packages/frontend/src/pages/Dashboard.tsx

```diff
@@ -725,9 +725,20 @@ const Dashboard: FC = () => {
 };
 
 const DashboardPage: FC = () => {
-    const { projectUuid } = useParams<{ projectUuid: string }>();
+    const { projectUuid, dashboardUuid } = useParams<{
+        projectUuid: string;
+        dashboardUuid: string;
+    }>();
     const { user } = useApp();
     const dashboardCommentsCheck = useDashboardCommentsCheck(user?.data);
+    const { clearDashboardStorage, getDashboardUuid } = useDashboardStorage();
+
+    useEffect(() => {
+        if (dashboardUuid !== getDashboardUuid()) {
+            // Clear storage if the dashboardUuid in storage doesn't match the current dashboardUuid
+            clearDashboardStorage();
+        }
+    }, [dashboardUuid, clearDashboardStorage, getDashboardUuid]);
```

Comment:
this prevents us from saving charts within dashboards. Perhaps a different check needs to be done?

Reply:
what about keying the storage on the dashboardUuid? So you have a cache per-id?

Reply:
We would still need to store the dashboardUuid separately so when we go from dashboard to create chart it knows from what cache to fetch. I propose we close this ticket as it is an edge case, we stopped deleting the charts in the DB and we are planning to change this UI flow.